core/lib/zksync_core/src/fee_model.rs

```diff
@@ -77,6 +79,59 @@ impl MainNodeFeeInputProvider {
     }
 }
 
+/// The fee model provider to be used in the API. It returns the maximal batch fee input between the projected main node one and
+/// the one from the last sealed miniblock.
+#[derive(Debug)]
+pub(crate) struct ApiFeeInputProvider {
+    inner: MainNodeFeeInputProvider,
+    connection_pool: ConnectionPool,
+}
+
+impl ApiFeeInputProvider {
+    pub fn new(
+        provider: Arc<dyn L1GasPriceProvider>,
+        config: FeeModelConfig,
+        connection_pool: ConnectionPool,
+    ) -> Self {
+        Self {
+            inner: MainNodeFeeInputProvider::new(provider, config),
+            connection_pool,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl BatchFeeModelInputProvider for ApiFeeInputProvider {
+    async fn get_batch_fee_input_scaled(
+        &self,
+        l1_gas_price_scale_factor: f64,
+        l1_pubdata_price_scale_factor: f64,
+    ) -> BatchFeeInput {
+        let inner_input = self
+            .inner
+            .get_batch_fee_input_scaled(l1_gas_price_scale_factor, l1_pubdata_price_scale_factor)
+            .await;
+        let last_miniblock_params = self
+            .connection_pool
+            .access_storage_tagged("api_fee_input_provider")
+            .await
+            .unwrap()
+            .blocks_dal()
+            .get_last_sealed_miniblock_header()
```

Comment:
Let's add a specialised method to DAL that loads only `batch_fee_input`

Reply:
Discussed in private not needed right now